### PR TITLE
[NEW] LC2432、2433、2434、2435: init

### PR DESCRIPTION
**Pull request by novayo**

### CODE REVIEW

File: 2432_The_Employee_That_Worked_on_the_Longest_Task/try_1.cpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Solution {
```
Comment (novayo) [inline 2432_The_Employee_That_Worked_on_the_Longest_Task/try_1.cpp:1]: Python version: [Link](https://github.com/novayo/LeetCode/blob/master/2432_The_Employee_That_Worked_on_the_Longest_Task/try_1.py)
```
class Solution:
    def hardestWorker(self, n: int, logs: List[List[int]]) -> int:
        ans = (logs[0][1], -logs[0][0])
        preTime = logs[0][1]
        
        for id ,time in logs[1:]:
            t = time - preTime
            ans = max(ans, (t, -id))
            preTime = time
            
        return -ans[1]
```

File: 2433_Find_The_Original_Array_of_Prefix_Xor/try_1.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#include<bitset>
```
Comment (novayo) [inline 2433_Find_The_Original_Array_of_Prefix_Xor/try_1.cpp:1]: Python version: [Link](https://github.com/novayo/LeetCode/blob/master/2433_Find_The_Original_Array_of_Prefix_Xor/try_1.py)
```
class Solution:
    def findArray(self, pref: List[int]) -> List[int]:
        
        def helper(a, b):
            # a ^ x = b
            x = ""
            for d1, d2 in zip(bin(a)[2:].zfill(32), bin(b)[2:].zfill(32)):
                if d2 == "1":
                    x += "1" if d1 == "0" else "0"
                else:
                    x += d1
            return int(x, 2)
            
        n = len(pref)
        
        ans = [pref[0]]
        for i in range(1, n):
            ans.append(helper(pref[i - 1], pref[i]))

        return 
```

File: 2434_Using_a_Robot_to_Print_the_Lexicographically_Smallest_String/try_1.cpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
+class Solution {
```
Comment (novayo) [inline 2434_Using_a_Robot_to_Print_the_Lexicographically_Smallest_String/try_1.cpp:1]: Python version: [Link](https://github.com/novayo/LeetCode/blob/master/2434_Using_a_Robot_to_Print_the_Lexicographically_Smallest_String/try_1.py)
```
from sortedcontainers import SortedList

class Solution:
    def robotWithString(self, s: str) -> str:
        
        bst = SortedList([ch for ch in s])

        ans = ""
        stack = []
        for ch in s:
            smallest = bst[0]

            while stack and stack[-1] <= smallest:
                ans += stack.pop()
            
            if ch == smallest:
                ans += ch
            else:
                stack.append(ch)
            bst.remove(ch)
                
        ans += "".join(stack[::-1])
        return 
```

File: 2435_Paths_in_Matrix_Whose_Sum_Is_Divisible_by_K/try_1.cpp [hunk 4]
```diff
@@ -0,0 +1,47 @@
+class Solution {
```
Comment (novayo) [inline 2435_Paths_in_Matrix_Whose_Sum_Is_Divisible_by_K/try_1.cpp:1]: Python version: [Link](https://github.com/novayo/LeetCode/blob/master/2435_Paths_in_Matrix_Whose_Sum_Is_Divisible_by_K/try_1.py)
```
class Solution:
    def numberOfPaths(self, grid: List[List[int]], k: int) -> int:
        
        m = len(grid)
        n = len(grid[0])
        
        MOD = k
        M = 10 ** 9 + 7
        
        dp = [[collections.Counter() for _ in range(n)] for _ in range(m)]
        
        dp[0][0][grid[0][0] % MOD] = 1
        for i in range(1, m):
            val = grid[i][0]
            for k, v in dp[i - 1][0].items():
                dp[i][0][(k + val) % MOD] += v % M
                
        for j in range(1, n):
            val = grid[0][j]
            for k, v in dp[0][j - 1].items():
                dp[0][j][(k + val) % MOD] += v % M
                
        for i in range(1, m):
            for j in range(1, n):
                val = grid[i][j]
                
                for k, v in dp[i][j - 1].items():
                    dp[i][j][(k + val) % MOD] += v % M
        
                for k, v in dp[i - 1][j].items():
                    dp[i][j][(k + val) % MOD] += v % M     
                    
        return dp[-1][-1][0] % M
```

File: 2433_Find_The_Original_Array_of_Prefix_Xor/try_1.cpp [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    }
+    
+public:
+    vector<int> findArray(vector<int>& pref) {
+        int n = pref.size();
+        vector<int> ans;
+        ans.push_back(pref[0]);
+        
+        for (int i=1; i < n; i++) {
+            ans.push_back(helper(pref[i-1], pref[i]));
```
Comment (Yabadado) [inline 2433_Find_The_Original_Array_of_Prefix_Xor/try_1.cpp:33]:
```suggestion
            ans.push_back(pref[i-1]^pref[i]);
```
寫完Truth table發現helper做的就是Xor 那可以用「^」來達成。

Comment (novayo) [reply 2433_Find_The_Original_Array_of_Prefix_Xor/try_1.cpp:33]: 是沒錯，不過目前的這些寫法是當下我跟阿寶一起討論出來的，我只是想把當下的做法翻譯成c++而已～

File: 2435_Paths_in_Matrix_Whose_Sum_Is_Divisible_by_K/try_1.cpp [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        
+        int MOD = k;
+        int M = 1e9 + 7;
+        
+        vector<vector<unordered_map<int, int>>> dp(m, vector<unordered_map<int, int>>(n, unordered_map<int, int>{}));
+        
+        dp[0][0][grid[0][0] % MOD] = 1;
+        for (int i=1; i<m; i++) {
+            int val = grid[i][0];
+            for (auto &item : dp[i-1][0]) {
```
Comment (Yabadado) [inline 2435_Paths_in_Matrix_Whose_Sum_Is_Divisible_by_K/try_1.cpp]:
```suggestion
            for (auto &[k, v] : dp[i-1][0]) {
```
unordered_map可以這樣取對應key value

File: 2433_Find_The_Original_Array_of_Prefix_Xor/try_1.cpp (hunk 5 above)
Comment (Yabadado) [reply 2433_Find_The_Original_Array_of_Prefix_Xor/try_1.cpp:33]: 了解!

File: 2435_Paths_in_Matrix_Whose_Sum_Is_Divisible_by_K/try_1.cpp (hunk 6 above)
Comment (novayo) [reply 2435_Paths_in_Matrix_Whose_Sum_Is_Divisible_by_K/try_1.cpp]: Done

**Review by Yabadado — APPROVED** [review]
Hen可以

**Pull request merged**